File: modulo2/shopapp/app/src/main/java/com/shopapp/presentation/ui/admin/dashboard/DashboardScreen.kt
// presentation/ui/admin/dashboard/DashboardScreen.kt
package com.shopapp.presentation.ui.admin.dashboard

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import com.shopapp.domain.model.OrderStatus
import com.shopapp.presentation.components.LoadingScreen
import com.shopapp.presentation.components.orderStatusColor
import com.shopapp.presentation.viewmodel.DashboardUiState
import com.shopapp.presentation.viewmodel.DashboardViewModel
import com.shopapp.theme.*
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

@Composable
fun DashboardScreen(
    onNavigate: (String) -> Unit,
    viewModel:  DashboardViewModel = hiltViewModel(),
) {
    val state       by viewModel.state.collectAsState()
    val lastUpdated by viewModel.lastUpdated.collectAsState()

    when (val s = state) {
        is DashboardUiState.Loading ->
            LoadingScreen("Cargando dashboard...")
        is DashboardUiState.Error   -> {
            Box(Modifier.fillMaxSize(), Alignment.Center) {
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Text("⚠️ ${s.message}", color = Error)
                    Spacer(Modifier.height(12.dp))
                    Button(onClick = viewModel::load,
                        colors = ButtonDefaults.buttonColors(containerColor = Accent)) {
                        Text("Reintentar", color = AccentOnDark)
                    }
                }
            }
        }
        is DashboardUiState.Success ->
            DashboardContent(
                stats       = s.stats,
                lastUpdated = lastUpdated,
                onNavigate  = onNavigate,
                onRefresh   = viewModel::load,
            )
    }
}

@Composable
private fun DashboardContent(
    stats:       com.shopapp.presentation.viewmodel.DashboardStats,
    lastUpdated: Long,
    onNavigate:  (String) -> Unit,
    onRefresh:   () -> Unit,
) {
    val timeFmt = SimpleDateFormat("HH:mm:ss", Locale.getDefault())
    val timeStr = if (lastUpdated > 0) timeFmt.format(Date(lastUpdated)) else "—"

    LazyColumn(
        modifier       = Modifier.fillMaxSize().background(Background),
        contentPadding = PaddingValues(16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp),
    ) {
        // Header
        item {
            Row(
                modifier              = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment     = Alignment.CenterVertically,
            ) {
                Column {
                    Text(
                        text       = "Dashboard",
                        style      = MaterialTheme.typography.headlineMedium,
                        fontWeight = FontWeight.Bold,
                        color      = TextPrimary,
                    )
                    Text(
                        text  = "Actualizado: $timeStr",
                        style = MaterialTheme.typography.bodySmall,
                        color = TextFaint,
                    )
                }
                IconButton(onClick = onRefresh) {
                    Icon(Icons.Default.Refresh, contentDescription = "Actualizar", tint = Accent)
                }
            }
        }

        // ── KPIs — fila 1 ─────────────────────────────────────
        item {
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                KpiCard(
                    title    = "Productos activos",
                    value    = stats.totalActiveProducts.toString(),
                    subtitle = if (stats.outOfStockProducts > 0)
                        "${stats.outOfStockProducts} sin stock" else null,
                    icon     = Icons.Default.Inventory,
                    color    = Accent,
                    hasAlert = stats.outOfStockProducts > 0,
                    onClick  = { onNavigate("admin/products") },
                    modifier = Modifier.weight(1f),
                )
                KpiCard(
                    title   = "Categorías activas",
                    value   = stats.activeCategories.toString(),
                    subtitle = "${stats.totalCategories} total",
                    icon    = Icons.Default.Category,
                    color   = Info,
                    onClick = { onNavigate("admin/categories") },
                    modifier = Modifier.weight(1f),
                )
            }
        }

        // ── KPIs — fila 2 ─────────────────────────────────────
        item {
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                KpiCard(
                    title    = "Pedidos totales",
                    value    = stats.totalOrders.toString(),
                    subtitle = if (stats.pendingOrders > 0)
                        "${stats.pendingOrders} pendientes" else null,
                    icon     = Icons.Default.ShoppingBag,
                    color    = Success,
                    hasAlert = stats.pendingOrders > 0,
                    onClick  = { onNavigate("admin/orders") },
                    modifier = Modifier.weight(1f),
                )
                KpiCard(
                    title    = "Usuarios activos",
                    value    = stats.activeUsers.toString(),
                    subtitle = "${stats.totalUsers} registrados",
                    icon     = Icons.Default.People,
                    color    = Warning,
                    onClick  = { onNavigate("admin/users") },
                    modifier = Modifier.weight(1f),
                )
            }
        }

        // ── KPIs financieros ──────────────────────────────────
        item {
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                KpiCard(
                    title    = "Facturación total",
                    value    = "$${"%.0f".format(stats.totalRevenue)}",
                    icon     = Icons.Default.TrendingUp,
                    color    = Accent,
                    modifier = Modifier.weight(1f),
                )
                KpiCard(
                    title    = "Precio medio",
                    value    = "$${"%.2f".format(stats.avgPrice)}",
                    subtitle = "por producto",
                    icon     = Icons.Default.Sell,
                    color    = TextSecondary,
                    modifier = Modifier.weight(1f),
                )
            }
        }

        // ── Pedidos por estado — gráfica de barras ─────────────
        if (stats.ordersByStatus.isNotEmpty()) {
            item {
                Surface(
                    color    = Surface,
                    shape    = MaterialTheme.shapes.large,
                    modifier = Modifier.fillMaxWidth(),
                ) {
                    Column(modifier = Modifier.padding(16.dp)) {
                        Row(
                            modifier              = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment     = Alignment.CenterVertically,
                        ) {
                            Text(
                                text       = "Pedidos por estado",
                                style      = MaterialTheme.typography.titleSmall,
                                fontWeight = FontWeight.Bold,
                                color      = TextPrimary,
                            )
                            TextButton(onClick = { onNavigate("admin/orders") }) {
                                Text("Ver todos", color = Accent,
                                    style = MaterialTheme.typography.bodySmall)
                            }
                        }
                        Spacer(Modifier.height(16.dp))

                        val total = stats.totalOrders.coerceAtLeast(1)
                        stats.ordersByStatus.entries.forEach { (statusValue, count) ->
                            val status = OrderStatus.fromValue(statusValue)
                            val color  = orderStatusColor(status)
                            val pct    = (count.toFloat() / total).coerceIn(0.02f, 1f)

                            Column(modifier = Modifier.padding(bottom = 10.dp)) {
                                Row(
                                    modifier              = Modifier.fillMaxWidth(),
                                    horizontalArrangement = Arrangement.SpaceBetween,
                                ) {
                                    Text(
                                        text  = status.label,
                                        style = MaterialTheme.typography.bodySmall,
                                        color = TextSecondary,
                                    )
                                    Text(
                                        text       = count.toString(),
                                        style      = MaterialTheme.typography.bodySmall,
                                        fontWeight = FontWeight.Bold,
                                        color      = color,
                                    )
                                }
                                Spacer(Modifier.height(4.dp))
                                Box(
                                    modifier = Modifier
                                        .fillMaxWidth()
                                        .height(7.dp)
                                        .background(Surface2, MaterialTheme.shapes.extraSmall),
                                ) {
                                    Box(
                                        modifier = Modifier
                                            .fillMaxWidth(pct)
                                            .fillMaxHeight()
                                            .background(color, MaterialTheme.shapes.extraSmall),
                                    )
                                }
                            }
                        }
                    }
                }
            }
        }

        // ── Productos con stock bajo ───────────────────────────
        item {
            Surface(
                color    = Surface,
                shape    = MaterialTheme.shapes.large,
                modifier = Modifier.fillMaxWidth(),
            ) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Row(
                        modifier              = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment     = Alignment.CenterVertically,
                    ) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(
                                Icons.Default.Warning,
                                contentDescription = null,
                                tint    = Warning,
                                modifier = Modifier.size(18.dp),
                            )
                            Spacer(Modifier.width(6.dp))
                            Text(
                                text       = "Stock bajo",
                                style      = MaterialTheme.typography.titleSmall,
                                fontWeight = FontWeight.Bold,
                                color      = TextPrimary,
                            )
                        }
                        TextButton(onClick = { onNavigate("admin/products") }) {
                            Text("Gestionar", color = Accent,
                                style = MaterialTheme.typography.bodySmall)
                        }
                    }

                    if (stats.lowStockProducts.isEmpty()) {
                        Box(
                            modifier         = Modifier.fillMaxWidth().padding(16.dp),
                            contentAlignment = Alignment.Center,
                        ) {
                            Text("✅ Sin alertas de stock", color = Success,
                                style = MaterialTheme.typography.bodySmall)
                        }
                    } else {
                        Spacer(Modifier.height(8.dp))
                        stats.lowStockProducts.forEach { product ->
                            Surface(
                                onClick  = { onNavigate("admin/products") },
                                color    = Surface2,
                                shape    = MaterialTheme.shapes.medium,
                                modifier = Modifier.fillMaxWidth().padding(bottom = 6.dp),
                            ) {
                                Row(
                                    modifier              = Modifier
                                        .fillMaxWidth()
                                        .padding(horizontal = 14.dp, vertical = 10.dp),
                                    horizontalArrangement = Arrangement.SpaceBetween,
                                    verticalAlignment     = Alignment.CenterVertically,
                                ) {
                                    Text(
                                        text     = product.name,
                                        style    = MaterialTheme.typography.bodySmall,
                                        color    = TextPrimary,
                                        fontWeight = FontWeight.Medium,
                                        modifier = Modifier.weight(1f),
                                        maxLines = 1,
                                    )
                                    Surface(
                                        color = if (product.stock == 0)
                                            Error.copy(alpha = 0.15f)
                                        else Warning.copy(alpha = 0.15f),
                                        shape = MaterialTheme.shapes.extraSmall,
                                    ) {
                                        Text(
                                            text       = if (product.stock == 0) "Agotado"
                                            else "${product.stock} uds.",
                                            color      = if (product.stock == 0) Error else Warning,
                                            fontSize   = 11.sp,
                                            fontWeight = FontWeight.Bold,
                                            modifier   = Modifier.padding(horizontal = 8.dp, vertical = 3.dp),
                                        )
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }

        // ── Acciones rápidas ──────────────────────────────────
        item {
            Surface(
                color    = Surface,
                shape    = MaterialTheme.shapes.large,
                modifier = Modifier.fillMaxWidth(),
            ) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Text(
                        text       = "⚡ Acciones rápidas",
                        style      = MaterialTheme.typography.titleSmall,
                        fontWeight = FontWeight.Bold,
                        color      = TextPrimary,
                        modifier   = Modifier.padding(bottom = 12.dp),
                    )
                    LazyRow(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                        items(listOf(
                            Triple("+ Categoría", Info,    "admin/categories"),
                            Triple("+ Producto",  Accent,  "admin/products"),
                            Triple("Ver pedidos", Success, "admin/orders"),
                            Triple("Usuarios",    Warning, "admin/users"),
                        )) { (label, color, route) ->
                            Surface(
                                onClick  = { onNavigate(route) },
                                color    = color.copy(alpha = 0.1f),
                                shape    = MaterialTheme.shapes.medium,
                            ) {
                                Text(
                                    text       = label,
                                    color      = color,
                                    fontWeight = FontWeight.Bold,
                                    style      = MaterialTheme.typography.bodySmall,
                                    modifier   = Modifier.padding(horizontal = 14.dp, vertical = 10.dp),
                                )
                            }
                        }
                    }
                }
            }
        }
    }
}

File: modulo4/flutter_shop_app/lib/presentation/screens/admin/dashboard_screen.dart
// lib/presentation/screens/admin/dashboard_screen.dart

import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import '../../../theme/app_colors.dart';
import '../../../core/utils/formatters.dart';
import '../../../domain/model/order.dart';
import '../../providers/dashboard_provider.dart';
import '../../widgets/kpi_card.dart';

class DashboardScreen extends ConsumerWidget {
  const DashboardScreen({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final state = ref.watch(dashboardProvider);

    return switch (state) {
      DashboardLoading() => const Center(
          child: CircularProgressIndicator(color: AppColors.accent),
        ),
      DashboardError(message: final msg) => Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              const Text('⚠️', style: TextStyle(fontSize: 48)),
              const SizedBox(height: 12),
              Text(msg, style: const TextStyle(color: AppColors.error),
                  textAlign: TextAlign.center),
              const SizedBox(height: 16),
              ElevatedButton(
                onPressed: () => ref.read(dashboardProvider.notifier).load(),
                child:     const Text('Reintentar'),
              ),
            ],
          ),
        ),
      DashboardSuccess(data: final d, loadedAt: final loadedAt) =>
          _DashboardContent(data: d, loadedAt: loadedAt),
    };
  }
}

class _DashboardContent extends ConsumerWidget {
  final DashboardData data;
  final DateTime      loadedAt;
  const _DashboardContent({required this.data, required this.loadedAt});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final timeFmt = '${loadedAt.hour.toString().padLeft(2,'0')}:'
                    '${loadedAt.minute.toString().padLeft(2,'0')}:'
                    '${loadedAt.second.toString().padLeft(2,'0')}';

    return RefreshIndicator(
      color:       AppColors.accent,
      onRefresh:   () => ref.read(dashboardProvider.notifier).load(),
      child: ListView(
        padding: const EdgeInsets.all(16),
        children: [

          // ── Header ────────────────────────────────────────
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text('Dashboard',
                      style: Theme.of(context).textTheme.headlineMedium),
                  Text('Actualizado: $timeFmt',
                      style: const TextStyle(color: AppColors.textFaint, fontSize: 11)),
                ],
              ),
              IconButton(
                icon:      const Icon(Icons.refresh_rounded, color: AppColors.accent),
                onPressed: () => ref.read(dashboardProvider.notifier).load(),
              ),
            ],
          ),
          const SizedBox(height: 20),

          // ── KPIs fila 1 ───────────────────────────────────
          Row(
            children: [
              Expanded(
                child: KpiCard(
                  title:    'Productos activos',
                  value:    '${data.totalActiveProducts}',
                  subtitle: data.outOfStockProducts > 0
                      ? '${data.outOfStockProducts} sin stock' : null,
                  icon:     Icons.inventory_2_outlined,
                  color:    AppColors.accent,
                  hasAlert: data.outOfStockProducts > 0,
                  onTap:    () => context.go('/admin/products'),
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: KpiCard(
                  title:    'Categorías activas',
                  value:    '${data.activeCategories}',
                  subtitle: '${data.totalCategories} total',
                  icon:     Icons.category_outlined,
                  color:    AppColors.info,
                  onTap:    () => context.go('/admin/categories'),
                ),
              ),
            ],
          ),
          const SizedBox(height: 12),

          // ── KPIs fila 2 ───────────────────────────────────
          Row(
            children: [
              Expanded(
                child: KpiCard(
                  title:    'Pedidos totales',
                  value:    '${data.totalOrders}',
                  subtitle: data.pendingOrders > 0
                      ? '${data.pendingOrders} pendientes' : null,
                  icon:     Icons.shopping_bag_outlined,
                  color:    AppColors.success,
                  hasAlert: data.pendingOrders > 0,
                  onTap:    () => context.go('/admin/orders'),
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: KpiCard(
                  title:    'Usuarios activos',
                  value:    '${data.activeUsers}',
                  subtitle: '${data.totalUsers} registrados',
                  icon:     Icons.people_outline,
                  color:    AppColors.warning,
                  onTap:    () => context.go('/admin/users'),
                ),
              ),
            ],
          ),
          const SizedBox(height: 12),

          // ── KPIs fila 3 — financieros ─────────────────────
          Row(
            children: [
              Expanded(
                child: KpiCard(
                  title:  'Facturación total',
                  value:  formatPrice(data.totalRevenue),
                  icon:   Icons.trending_up_rounded,
                  color:  AppColors.accent,
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: KpiCard(
                  title:    'Precio medio',
                  value:    formatPrice(data.avgPrice),
                  subtitle: 'por producto',
                  icon:     Icons.sell_outlined,
                  color:    AppColors.textSecondary,
                ),
              ),
            ],
          ),
          const SizedBox(height: 20),

          // ── Pedidos por estado — barras ───────────────────
          if (data.ordersByStatus.isNotEmpty) ...[
            _SectionCard(
              title:     'Pedidos por estado',
              actionLabel: 'Ver todos',
              onAction:  () => context.go('/admin/orders'),
              child: Column(
                children: data.ordersByStatus.entries.map((entry) {
                  final status  = OrderStatus.fromValue(entry.key);
                  final count   = entry.value;
                  final total   = data.totalOrders.clamp(1, 999999);
                  final pct     = (count / total).clamp(0.02, 1.0);
                  final color   = orderStatusColor(status);

                  return Padding(
                    padding: const EdgeInsets.only(bottom: 10),
                    child: Column(
                      children: [
                        Row(
                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
                          children: [
                            Text(status.label,
                                style: const TextStyle(
                                  color: AppColors.textSecondary, fontSize: 12,
                                )),
                            Text(
                              '$count',
                              style: TextStyle(
                                color: color, fontWeight: FontWeight.bold, fontSize: 12,
                              ),
                            ),
                          ],
                        ),
                        const SizedBox(height: 4),
                        ClipRRect(
                          borderRadius: BorderRadius.circular(3),
                          child: LinearProgressIndicator(
                            value:     pct,
                            minHeight: 7,
                            backgroundColor:  AppColors.surface2,
                            valueColor: AlwaysStoppedAnimation(color),
                          ),
                        ),
                      ],
                    ),
                  );
                }).toList(),
              ),
            ),
            const SizedBox(height: 14),
          ],

          // ── Stock bajo ────────────────────────────────────
          _SectionCard(
            title:      'Stock bajo',
            titleIcon:  Icons.warning_amber_rounded,
            titleColor: AppColors.warning,
            actionLabel:'Gestionar',
            onAction:   () => context.go('/admin/products'),
            child: data.lowStockProducts.isEmpty
                ? const Center(
                    child: Padding(
                      padding: EdgeInsets.symmetric(vertical: 8),
                      child:   Text(
                        '✅ Sin alertas de stock',
                        style: TextStyle(color: AppColors.success, fontSize: 13),
                      ),
                    ),
                  )
                : Column(
                    children: data.lowStockProducts.map((p) {
                      final isOut = p.stock == 0;
                      return GestureDetector(
                        onTap: () => context.go('/admin/products'),
                        child: Container(
                          margin:  const EdgeInsets.only(bottom: 6),
                          padding: const EdgeInsets.symmetric(
                            horizontal: 12, vertical: 10,
                          ),
                          decoration: BoxDecoration(
                            color:        AppColors.surface2,
                            borderRadius: BorderRadius.circular(10),
                          ),
                          child: Row(
                            mainAxisAlignment: MainAxisAlignment.spaceBetween,
                            children: [
                              Expanded(
                                child: Text(
                                  p.name,
                                  style: const TextStyle(
                                    color:    AppColors.textPrimary,
                                    fontSize: 13,
                                  ),
                                  overflow: TextOverflow.ellipsis,
                                ),
                              ),
                              Container(
                                padding: const EdgeInsets.symmetric(
                                  horizontal: 8, vertical: 2,
                                ),
                                decoration: BoxDecoration(
                                  color:        isOut
                                      ? AppColors.error.withValues(alpha: 0.12)
                                      : AppColors.warning.withValues(alpha: 0.12),
                                  borderRadius: BorderRadius.circular(6),
                                ),
                                child: Text(
                                  isOut ? 'Agotado' : '${p.stock} uds.',
                                  style: TextStyle(
                                    color:      isOut ? AppColors.error : AppColors.warning,
                                    fontSize:   11,
                                    fontWeight: FontWeight.bold,
                                  ),
                                ),
                              ),
                            ],
                          ),
                        ),
                      );
                    }).toList(),
                  ),
          ),
          const SizedBox(height: 14),

          // ── Acciones rápidas ──────────────────────────────
          _SectionCard(
            title: '⚡ Acciones rápidas',
            child: Wrap(
              spacing: 8, runSpacing: 8,
              children: [
                ('+ Categoría', AppColors.info,    '/admin/categories'),
                ('+ Producto',  AppColors.accent,  '/admin/products'),
                ('Ver pedidos', AppColors.success,  '/admin/orders'),
                ('Usuarios',    AppColors.warning,  '/admin/users'),
              ].map((item) => GestureDetector(
                onTap: () => context.go(item.$3),
                child: Container(
                  padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 8),
                  decoration: BoxDecoration(
                    color:        item.$2.withValues(alpha: 0.1),
                    borderRadius: BorderRadius.circular(10),
                    border:       Border.all(color: item.$2.withValues(alpha: 0.3)),
                  ),
                  child: Text(
                    item.$1,
                    style: TextStyle(
                      color:      item.$2,
                      fontWeight: FontWeight.bold,
                      fontSize:   13,
                    ),
                  ),
                ),
              )).toList(),
            ),
          ),
          const SizedBox(height: 32),
        ],
      ),
    );
  }
}

class _SectionCard extends StatelessWidget {
  final String       title;
  final IconData?    titleIcon;
  final Color?       titleColor;
  final String?      actionLabel;
  final VoidCallback? onAction;
  final Widget       child;

  const _SectionCard({
    required this.title,
    required this.child,
    this.titleIcon,
    this.titleColor,
    this.actionLabel,
    this.onAction,
  });

  @override
  Widget build(BuildContext context) => Container(
    width:   double.infinity,
    padding: const EdgeInsets.all(16),
    decoration: BoxDecoration(
      color:        AppColors.surface,
      borderRadius: BorderRadius.circular(16),
    ),
    child: Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: [
            Row(
              children: [
                if (titleIcon != null) ...[
                  Icon(titleIcon, color: titleColor ?? AppColors.textPrimary, size: 18),
                  const SizedBox(width: 6),
                ],
                Text(
                  title,
                  style: TextStyle(
                    color:      titleColor ?? AppColors.textPrimary,
                    fontWeight: FontWeight.bold,
                    fontSize:   15,
                  ),
                ),
              ],
            ),
            if (actionLabel != null && onAction != null)
              GestureDetector(
                onTap: onAction,
                child: Text(
                  '$actionLabel →',
                  style: const TextStyle(
                    color: AppColors.accent, fontSize: 12, fontWeight: FontWeight.w600,
                  ),
                ),
              ),
          ],
        ),
        const SizedBox(height: 16),
        child,
      ],
    ),
  );
}